basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/LTIRoleMapperImpl.java

```diff
@@ -73,6 +58,26 @@ public void setSiteService(SiteService siteService) {
             throw new LTIException( "launch.site.user.missing", "user_id="+user.getId()+ ", siteId="+site.getId(), null);
         }
 
+        String ltiRole = null;
+
+        if (trustedConsumer) {
+            // If the launch is from a trusted consumer, just return the user's
+            // role in the site. No need to map.
+            Member member = site.getMember(user.getId());
```

Comment:
If the user isn't a member of the site (eg access a site through .anon) then member will be null and this would then NPE on the following line.